src/components/shared/CallNodeContextMenu.js

```diff
@@ -31,22 +30,13 @@ import {
 import type { TransformType } from '../../types/transforms';
 import type { ImplementationFilter } from '../../types/actions';
 import type { TabSlug } from '../../app-logic/tabs-handling';
-import type {
-  IndexIntoCallNodeTable,
-  CallNodeInfo,
-  CallNodePath,
-} from '../../types/profile-derived';
-import type { Thread, ThreadIndex } from '../../types/profile';
 import type { ConnectedProps } from '../../utils/connect';
+import type { RightClickedCallNodeInfo } from '../../selectors/right-clicked-call-node';
 
 type StateProps = {|
-  +thread: Thread,
-  +threadIndex: ThreadIndex,
-  +callNodeInfo: CallNodeInfo,
+  +rightClickedCallNode: RightClickedCallNodeInfo | null,
```

Comment:
same remark as for the marker context menu: maybe call this `rightClickedCallNodeInfo` so that in the code we better know the type of the variables we're dealing with.